src/game/movement.rs

```diff
@@ -148,9 +151,28 @@ fn update_roll_timer(time: Res<Time>, mut query: Query<(&mut Roll, &mut GridMove
     let dt = time.delta_seconds();
     for (mut roll, mut movement) in query.iter_mut() {
         roll.timer.tick(Duration::from_secs_f32(dt));
-        if roll.timer.finished() {
+
+        if !roll.timer.finished() && movement.is_rolling{
             movement.is_rolling = false;
             roll.timer.reset();
         }
     }
 }
+
+fn update_cooldown_timer(time: Res<Time>, mut query: Query<(&mut Roll, &mut GridMovement)>) {
+    let dt = Duration::from_secs_f32(time.delta_seconds());
+    for (mut roll, movement) in query.iter_mut() {
+
+        if roll.cooldown.finished() {
+            roll.cooldown.paused();
+        } else {
+            roll.cooldown.tick(dt);
+        }
```

Comment:
The thing about bevy timers is they don't automatically count unless you tick() them with a duration, usually your delta time

So this code will never actually tick the cooldown timer if i'm reading it right (since the timer can never finish() if it hasn't been tick()'ed, but the tick() is behind a finish() if condition)

I think the functionality we need can be accomplished without the use of pause/unpause also, something like:


- always be ticking the cooldown timer with dt
- while rolling, also be ticking the roll timer with dt
- when a roll finishes, reset both timers
- before starting a new roll in the input manager code, check that the cooldown timer is in a finished() state